### PR TITLE
feat(spinner): add isSVG

### LINKED ISSUE #5343: Spinner - SVG variation
https://github.com/patternfly/patternfly/pull/3690

We added an SVG variation to the spinner.

* Uses an SVG instead of divs
* Much better performance
* Spins in the correct direction

cc @mcarrano

### PR DESCRIPTION
**Pull request by redallen**
**What**: Closes #5343

**Additional issues**:

### CODE REVIEW

**Review by KKoukiou — DISMISSED** [review]
Do you plan to set the isSVG by default to 'true' at some point? I guess you consider it a breaking change but should be add a reminder in the code to set it by default to 'true' in the future?

**Comment by redallen** [comment]
I personally think it should be the default. @mattnolting do you think `isSVG` should be the default?

**Comment by mattnolting** [comment]
> I personally think it should be the default. @mattnolting do you think `isSVG` should be the default?

Yes, I do

File: packages/react-core/src/components/Spinner/Spinner.tsx [hunk 1]
```diff
@@ -16,24 +16,42 @@ export interface SpinnerProps extends Omit<React.HTMLProps<HTMLSpanElement>, 'si
...
+}: SpinnerProps) => {
+  const Component = isSVG ? 'svg' : ('span' as any);
+
+  return (
+    <Component
+      className={css(styles.spinner, styles.modifiers[size], className)}
+      role="progressbar"
+      aria-valuetext={ariaValueText}
+      {...(isSVG && { viewBox: '0 0 100 100' })}
+      {...(isSVG && diameter && { style: { '--pf-c-spinner--diameter': diameter } })}
```
Comment (mcoker) [inline packages/react-core/src/components/Spinner/Spinner.tsx:42]: `--pf-c-spinner--diameter` works with regular spinners, too. We just didn't have an example of it.

Comment (redallen) [reply packages/react-core/src/components/Spinner/Spinner.tsx]: Fixed!

**Review by mcoker — APPROVED** [review]
L🎉TM!

**Comment by tlabaj** [comment]
> Do you plan to set the isSVG by default to 'true' at some point? I guess you consider it a breaking change but should be add a reminder in the code to set it by default to 'true' in the future?

@KKoukiou yes, that would be a breaking change.  We can open an issue and mark it breaking change issue and plan to pull it in for our next major release.

**Review by kmcfaul — APPROVED** [review]

File: packages/react-core/src/components/Spinner/examples/Spinner.md [hunk 2]
```diff
@@ -20,9 +20,18 @@ import React from 'react';
...
-  <Spinner size="lg"/>
-  <Spinner size="xl"/>
+  <Spinner isSVG size="sm"/>
+  <Spinner isSVG size="md"/>
+  <Spinner isSVG size="lg"/>
+  <Spinner isSVG size="xl"/>
 </React.Fragment>
 ```
+
+### Custom size (SVG only)
```
Comment (tlabaj) [inline packages/react-core/src/components/Spinner/examples/Spinner.md:30]: based on @mcoker 's comment, you can remove the `(SVG only)` text.

**Review by tlabaj — CHANGES_REQUESTED** [review]
Can you also add the new props to the demo app and add integration tests.

File: packages/react-core/src/components/Spinner/examples/Spinner.md (hunk 2 above)
Comment (redallen) [reply packages/react-core/src/components/Spinner/examples/Spinner.md]: Done.

**Comment by tlabaj** [comment]
@redallen @KKoukiou I opened a breaking change issue #5404  to default the isSVG to true.

File: packages/react-integration/demo-app-ts/src/components/demos/SpinnerDemo/SpinnerDemo.tsx [hunk 3]
```diff
@@ -10,13 +10,13 @@ export class SpinnerDemo extends React.Component<SpinnerProps> {
   render() {
```
Comment (tlabaj) [inline packages/react-integration/demo-app-ts/src/components/demos/SpinnerDemo/SpinnerDemo.tsx:10]: You should also add the diameter prop. and an integration test.

Comment (redallen) [reply packages/react-integration/demo-app-ts/src/components/demos/SpinnerDemo/SpinnerDemo.tsx:10]: Done.

**Review by jessiehuff — APPROVED** [review]
LGTM!

**Review by kmcfaul — APPROVED** [review]

**Review by tlabaj — APPROVED** [review]
looks great!

**Pull request merged**